interface/java_binding/src/main/java/org/khronos/ktx/VkFormat.java

```diff
@@ -0,0 +1,384 @@
+    public static final int VK_FORMAT_BC6H_UFLOAT_BLOCK = 143;
+    public static final int VK_FORMAT_BC6H_SFLOAT_BLOCK = 144;
+    public static final int VK_FORMAT_BC7_UNORM_BLOCK = 145;
+    public static final int VK_FORMAT_BC7_SRGB_BLOCK = 146;
+    public static final int VK_FORMAT_ETC2_R8G8B8_UNORM_BLOCK = 147;
+    public static final int VK_FORMAT_ETC2_R8G8B8_SRGB_BLOCK = 148;
+    public static final int VK_FORMAT_ETC2_R8G8B8A1_UNORM_BLOCK = 149;
+    public static final int VK_FORMAT_ETC2_R8G8B8A1_SRGB_BLOCK = 150;
+    public static final int VK_FORMAT_ETC2_R8G8B8A8_UNORM_BLOCK = 151;
+    public static final int VK_FORMAT_ETC2_R8G8B8A8_SRGB_BLOCK = 152;
+    public static final int VK_FORMAT_EAC_R11_UNORM_BLOCK = 153;
+    public static final int VK_FORMAT_EAC_R11_SNORM_BLOCK = 154;
+    public static final int VK_FORMAT_EAC_R11G11_UNORM_BLOCK = 155;
+    public static final int VK_FORMAT_EAC_R11G11_SNORM_BLOCK = 156;
+    public static final int VK_FORMAT_ASTC_4x4_UNORM_BLOCK = 157;
+    public static final int VK_FORMAT_ASTC_4x4_SRGB_BLOCK = 158;
+    public static final int VK_FORMAT_ASTC_5x4_UNORM_BLOCK = 159;
+    public static final int VK_FORMAT_ASTC_5x4_SRGB_BLOCK = 160;
+    public static final int VK_FORMAT_ASTC_5x5_UNORM_BLOCK = 161;
+    public static final int VK_FORMAT_ASTC_5x5_SRGB_BLOCK = 162;
+    public static final int VK_FORMAT_ASTC_6x5_UNORM_BLOCK = 163;
+    public static final int VK_FORMAT_ASTC_6x5_SRGB_BLOCK = 164;
+    public static final int VK_FORMAT_ASTC_6x6_UNORM_BLOCK = 165;
+    public static final int VK_FORMAT_ASTC_6x6_SRGB_BLOCK = 166;
+    public static final int VK_FORMAT_ASTC_8x5_UNORM_BLOCK = 167;
+    public static final int VK_FORMAT_ASTC_8x5_SRGB_BLOCK = 168;
+    public static final int VK_FORMAT_ASTC_8x6_UNORM_BLOCK = 169;
+    public static final int VK_FORMAT_ASTC_8x6_SRGB_BLOCK = 170;
+    public static final int VK_FORMAT_ASTC_8x8_UNORM_BLOCK = 171;
+    public static final int VK_FORMAT_ASTC_8x8_SRGB_BLOCK = 172;
+    public static final int VK_FORMAT_ASTC_10x5_UNORM_BLOCK = 173;
+    public static final int VK_FORMAT_ASTC_10x5_SRGB_BLOCK = 174;
+    public static final int VK_FORMAT_ASTC_10x6_UNORM_BLOCK = 175;
+    public static final int VK_FORMAT_ASTC_10x6_SRGB_BLOCK = 176;
+    public static final int VK_FORMAT_ASTC_10x8_UNORM_BLOCK = 177;
+    public static final int VK_FORMAT_ASTC_10x8_SRGB_BLOCK = 178;
+    public static final int VK_FORMAT_ASTC_10x10_UNORM_BLOCK = 179;
+    public static final int VK_FORMAT_ASTC_10x10_SRGB_BLOCK = 180;
+    public static final int VK_FORMAT_ASTC_12x10_UNORM_BLOCK = 181;
+    public static final int VK_FORMAT_ASTC_12x10_SRGB_BLOCK = 182;
+    public static final int VK_FORMAT_ASTC_12x12_UNORM_BLOCK = 183;
+    public static final int VK_FORMAT_ASTC_12x12_SRGB_BLOCK = 184;
+    // Provided by public static final int VK_VERSION_1_1
+    public static final int VK_FORMAT_G8B8G8R8_422_UNORM = 1000156000;
```

Comment:
Hi, I looked at the AWK script and I have NO idea how to modify it to generate a `VkFormat.java` file :). If you guide me, I might be able to do it.

Reply:
I looked again at the generated vkformat_enum.h. It is used by the .c file that checks for prohibited Vulkan formats so it needs all the formats. From what you said in the main discussion thread, this class is used to set the vkFormat field of CreateInfo so it should not include any prohibited formats.

At a minimum those formats need to be removed from this file. Alternatively we can make a script - I don't think it should be added to mkvkformatfiles because its current outputs go in the same directory as it - to generate the file from either lib/vkformat_enums.h or from vulkan_core.h. What do you think?

Reply:
Can you describe how I can recognize which formats are prohibited?